build/platform-android.mk

```diff
@@ -37,16 +37,23 @@ $(error TARGET is not set)
 endif
 
 TOOLCHAINPREFIX = $(shell NDK_PROJECT_PATH=$(SRC_PATH)/codec/build/android/dec make --no-print-dir -f $(NDKROOT)/build/core/build-local.mk DUMP_TOOLCHAIN_PREFIX APP_ABI=$(APP_ABI))
+TOOLCHAIN_NAME = $(shell NDK_PROJECT_PATH=$(SRC_PATH)/codec/build/android/dec make --no-print-dir -f $(NDKROOT)/build/core/build-local.mk DUMP_TOOLCHAIN_NAME APP_ABI=$(APP_ABI))
+GCC_TOOLCHAIN_PATH = $(shell dirname $(TOOLCHAINPREFIX) | xargs dirname )
 
 SYSROOT = $(NDKROOT)/platforms/android-$(NDKLEVEL)/arch-$(ARCH)
 CXX = $(TOOLCHAINPREFIX)g++
 CC = $(TOOLCHAINPREFIX)gcc
 AR = $(TOOLCHAINPREFIX)ar
 CFLAGS += -DANDROID_NDK -fpic --sysroot=$(SYSROOT) -MMD -MP
+CFLAGS += -isystem $(NDKROOT)/sysroot/usr/include -isystem $(NDKROOT)/sysroot/usr/include/$(TOOLCHAIN_NAME)
```

Comment:
Should we maybe only add this if `NDK_TOOLCHAIN_VERSION` is `clang`? On the other hand, I think it might be harmless to add on older NDK versions even if this directory doesn't exist.

As for the second directory added, I don't see any `sysroot/usr/include/clang` directory anywhere in my NDK - does that really exist in some version, or is this a mistake?

Reply:
thanks for your comments.
for your questions:
1. I think it is harmless to add it for older NDK version. (when NDK version is greater then r14, all the header files have been moved to this two path: a. $NDK/sysroot/usr/include b.$NDK/sysroot/usr/include/$(TOOLCHAIN_NAME))
2. $(TOOLCHAIN_NAME) is not equal to $(TOOLCHAIN_VERSION) for NDK. the value of $(TOOLCHAIN_NAME) is:
arm : arm-Linux-androideabi
arm64: aarch64-linux-android
x86: i686-linux-android
x86_64: x86_64-linux-android

Reply:
Ah, I see - I didn't notice the difference between `$(NDK_TOOLCHAIN_VERSION)` and `$(TOOLCHAIN_NAME)` here. So these issues I raised are ok then.

So I presume that `$(NDK_TOOLCHAIN_VERSION)` is something that the user needs to manually set by passing `NDK_TOOLCHAIN_VERSION=clang` when compiling, right? That's probably ok, but is there possibly any way to find out what is the default for that also in the same way as we query the other values? If not this is fine as is.

Reply:
Ah, I see - I didn't notice the difference between `$(NDK_TOOLCHAIN_VERSION)` and `$(TOOLCHAIN_NAME)` here. So these issues I raised are ok then.

So I presume that `$(NDK_TOOLCHAIN_VERSION)` is something that the user needs to manually set by passing `NDK_TOOLCHAIN_VERSION=clang` when compiling, right? That's probably ok, but is there possibly any way to find out what is the default for that also in the same way as we query the other values? If not this is fine as is.

Reply:
yes. the user should manually pass NDK_TOOLCHAIN_VERSION=clang when compiling. 
of course, we can query this value and set it as the default value. But I think this is not a good choice for some versions. e.g. for NDK-r14, the default value of NDK_TOOLCHAIN_VERSION is clang, but they still use GCC toolchain as the default tool. Maybe we can take this choice for higher NDK version(e.g. NDK version greater than r17), but I think it is not a good choice for right now.

Reply:
Ok, fair enough - then I'm ok with this change as it is right now. Thanks for taking the time to explain your choices!

Reply:
Thank you for taking the time to review this code.